cmake/FindJansson.cmake

```diff
@@ -0,0 +1,15 @@
+if (Jansson_INCLUDE_DIRS AND Jansson_LIBRARY AND Jansson_VERSION)
+  set(Jansson_FOUND TRUE)
+else ()
+  find_path(Jansson_INCLUDE_DIRS NAMES jansson.h)
+  find_library(Jansson_LIBRARY NAMES jansson)
+
+  if (Jansson_INCLUDE_DIRS AND Jansson_LIBRARY)
+      set(regex_jansson_version "#define[ \t]+JANSSON_VERSION[ \t]+[\"]([^\"]+)[\"]")
+      file(STRINGS "${Jansson_INCLUDE_DIRS}/jansson.h" jansson_version REGEX "${regex_jansson_version}")
+      string(REGEX REPLACE "${regex_jansson_version}" "\\1" Jansson_VERSION "${jansson_version}")
```

Comment:
What is replaced here and to what? Can it be done in one line above?

Reply:
On line `9` the `file(STRINGS ...` does not have a feature to read in parts of a line from a file, so it has to read in a whole line from a file, the whole line that matches the regex.
On line `10` from the whole line of `#define JANSSON_VERSION "2.14"` we want to further separate out the version,
and put that into the version variable, and thats what we do by using the first capture group of the regex.

I am not aware of any built in feature which can read in only parts of a file based on a regex capture group.